src/app/qgisapp.cpp

```diff
@@ -13301,3 +13328,104 @@ void QgisApp::transactionGroupCommitError( const QString &error )
 {
   displayMessage( tr( "Transaction" ), error, QgsMessageBar::CRITICAL );
 }
+
+QgsFeature QgisApp::duplicateFeatures( QgsMapLayer *mlayer, const QgsFeature &feature )
+{
+  if ( mlayer->type() != QgsMapLayer::VectorLayer )
+    return QgsFeature();
+
+  QgsVectorLayer *layer = qobject_cast<QgsVectorLayer *>( mlayer );
+
+  layer->startEditing();
```

Comment:
What do you think about checking that `layer` is not null in this case (just after casting `mLayer` in a vector layer)?

Reply:
Sorry, I didn't see that you previously checked the type of layer...